eng/common/pipelines/templates/steps/docs-metadata-release.yml

```diff
@@ -30,14 +30,19 @@ steps:
       Write-Host "This script is not executing on Windows, skipping registry modification."
     }
   displayName: Enable Long Paths if Necessary
-  
-- pwsh: |
-    git clone https://github.com/${{ parameters.TargetDocRepoOwner }}/${{ parameters.TargetDocRepoName }} ${{ parameters.WorkingDirectory }}/repo
-  displayName: Clone Documentation Repository
-  ignoreLASTEXITCODE: false
+
+- template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+  parameters:
+    SkipDefaultCheckout: true
+    Repositories:
+      - Name: ${{ parameters.TargetDocRepoOwner }}/${{ parameters.TargetDocRepoName }}
+        Commitish: master
+        WorkingDirectory: ${{ parameters.WorkingDirectory }}/repo
+    Paths: ${{ parameters.Paths }}
```

Comment:
Unless I'm missing something and you don't add paths to the callers of this template this sparse checkout isn't going to do anything which seems incorrect.

Reply:
`Paths` needs to be passed in now from the specific languages, as you suggested [here](https://github.com/Azure/azure-sdk-tools/pull/1585#discussion_r623445138).

Reply:
So by default when paths is empty, sparse checkout will just checkout `eng`.

Reply:
I think the alternative is to change the parameter definition format to the `name: / value: / default: ` syntax so that I can exclude the default which will mark it as a required parameter (meaning you can't include this template without specifying it).

Reply:
Ok so how do we stage this then?

Reply:
Hmm, perhaps we should just default `Paths` to `/` to handle repos where we wouldn't be passing in overrides anyway.

Reply:
@weshaggard updated to default to the old clone behavior if `SparseCheckoutPaths` is not specified.